### PR TITLE
Feat/stations

### PR DESCRIPTION
**Pull request by Arthurchkeee**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution

image
image
image
image

#### 🎯 Related issues

[_Links to related issues_
](https://angularteam.atlassian.net/browse/SCRUM-13)

### CODE REVIEW

File: src/app/core/store/user/user.reducer.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
 } from './user.actions';
 
 const initialUserState: UserState = {
-  currentAccess: 'guest',
+  currentAccess: 'manager',
```
Comment (NataRep) [inline src/app/core/store/user/user.reducer.ts]: This line must have been left over from testing and needs to be changed back.

**Review by NataRep — CHANGES_REQUESTED** [review]
Great job! I left one comment, check it out

File: src/app/features/admin/services/popup.service.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  providedIn: 'root',
+})
+export class PopUpService {
+
+  makeCapitalPopup(title:string): string {
+    return (
+      `` +
+      `<div>Station: ${title}</div>`
+    );
+  }
```
Comment (mogitrash) [inline src/app/features/admin/services/popup.service.ts]: Why we need to use this service instead of snackbars in Angular Material?

**Review by mogitrash — CHANGES_REQUESTED** [review]
Nice job! Have some questions

File: src/app/features/admin/services/popup.service.ts (hunk 2 above)
Comment (Arthurchkeee) [reply src/app/features/admin/services/popup.service.ts]: It's not about snackbars. It's for showing city name cloud from second screenshot

File: src/app/core/store/user/user.reducer.ts (hunk 1 above)
Comment (Arthurchkeee) [reply src/app/core/store/user/user.reducer.ts]: Yeah, I forget about this 😬

File: src/app/pages/admin-page/admin-page.component.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
+.map-container {
+  width: 100%;
+  height: 600px;
+  z-index: 0;
+}
+
+.content-container {
+  display: grid;
+  grid-template-columns: 3fr 1fr;
+  gap: 16px; /* Опционально, для отступа между элементами */
```
Comment (Aleksa013) [inline src/app/pages/admin-page/admin-page.component.scss]: I think it's better to use English for comments

File: src/app/pages/admin-page/admin-page.component.ts [hunk 4]
```diff
@@ -1,10 +1,321 @@
...
+        Validators.required,
+        this.forbidSelectedValues(selectedValues),
+      ]);
+      control.updateValueAndValidity();
+    });
+  }
+
+  private resetRelations() {
+    const relationsArray = this.stationForm.get('relations') as FormArray;
+    while (relationsArray.length !== 0) {
```
Comment (Aleksa013) [inline src/app/pages/admin-page/admin-page.component.ts]: Here can use
while (relationsArray.length)

**Review by Aleksa013 — APPROVED** [review]
Great job!
But you need to fix  UserInitialState (commit from Natalia)

File: src/app/pages/admin-page/admin-page.component.ts (hunk 4 above)
Comment (Arthurchkeee) [reply src/app/pages/admin-page/admin-page.component.ts]: Thanks, I fixed that and found, that I missed unsubscription, when I reset form

**Review by NataRep — APPROVED** [review]

File: src/app/pages/admin-page/admin-page.component.html [hunk 5]
```diff
@@ -1 +1,88 @@
...
+      }
+      <mat-paginator
+        [pageSizeOptions]="[5, 10, 20]"
+        showFirstLastButtons
+        aria-label="Select page of periodic elements"
+      >
+      </mat-paginator>
+    }
+  </mat-card-content>
+</mat-card>
```
Comment (NataRep) [inline src/app/pages/admin-page/admin-page.component.html]: Please move the stations to a separate component to insert into the Admin Panel page.

File: src/app/pages/admin-page/admin-page.component.scss [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(255, 255, 255, 0.8);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 1000;
+}
```
Comment (NataRep) [inline src/app/pages/admin-page/admin-page.component.scss]: and styles, too.

File: src/app/pages/admin-page/admin-page.component.ts [hunk 7]
```diff
@@ -1,8 +1,290 @@
...
+    this.store.dispatch(canDelete({ station, coordinates }));
+  }
+
+  protected findStationNameById(id: number, stations: Station[]): string {
+    const foundStation = stations.find((station) => {
+      return station.id === id;
+    });
+    return foundStation ? foundStation.city : 'Station not found';
+  }
+}
```
Comment (NataRep) [inline src/app/pages/admin-page/admin-page.component.ts]: Please move the Stations to a separate component to insert into the Admin Panel page.

**Review by NataRep — CHANGES_REQUESTED** [review]

File: src/app/pages/admin-page/admin-page.component.ts (hunk 7 above)
Comment (Arthurchkeee) [reply src/app/pages/admin-page/admin-page.component.ts]: Okay, i fixed that

**Review by AccountLoan — APPROVED** [review]
👍

**Review by mogitrash — APPROVED** [review]
Nice job 🥇

**Review by NataRep — APPROVED** [review]

**Pull request merged**